fixtures/tests/_fixtures/test_warnings.py

```diff
@@ -47,3 +59,60 @@ def test_capture_category(self):
         for i, category in enumerate(categories):
             c = w.captures[i]
             self.assertEqual(category, c.category)
+
+
+class TestWarningsFilter(testtools.TestCase, fixtures.TestWithFixtures):
+
+    def test_filter(self):
+        fixture = fixtures.WarningsFilter(
+            [
+                {
+                    'action': 'ignore',
+                    'category': DeprecationWarning,
+                },
+                {
+                    'action': 'once',
+                    'category': UserWarning,
+                },
+            ],
+        )
+        self.useFixture(fixture)
+        with warnings.catch_warnings(record=True) as w:
+            warnings.warn('deprecated', DeprecationWarning)
+            warnings.warn('user', UserWarning)
+
+        # only the user warning should be present, and it should only have been
+        # raised once
+        self.assertEqual(1, len(w))
+
+    def test_filters_restored(self):
+
+        class CustomWarning(Warning):
+            pass
+
+        fixture = fixtures.WarningsFilter(
+            [
+                {
+                    'action': 'once',
+                    'category': CustomWarning,
+                },
+            ],
+        )
+
+        # we copy the filter values rather than a reference to the containing
+        # list since that can change
+        old_filters = warnings.filters[:]
+
+        # NOTE: we intentionally do not use 'self.useFixture' since we want to
+        # teardown the fixture manually here before we exit this test method
```

Comment:
Fair enough, but consider `with fixture:` instead so that the fixture is still cleaned up if the test fails inside the section of code where it's active.

Reply:
Didn't know that was a thing. Will respin once I get a resolution on the below.

Reply:
Done